cabal-install/src/Distribution/Client/ProjectPlanning/Types.hs

```diff
@@ -542,7 +528,7 @@ elabDistDirParams shared elab =
         ElabPackage _ -> Nothing
     , distParamCompilerId = compilerId (pkgConfigCompiler shared)
     , distParamPlatform = pkgConfigPlatform shared
-    , distParamOptimization = elabOptimization elab
+    , distParamOptimization = LBC.withOptimization $ elabBuildOptions elab
```

Comment:
I find it very annoying that we have this here. It's because we use different directories depending on the optimization level. Do we really need to?

Reply:
Do you want to file a separate ticket for that discussion?

Reply:
I will when I get back to work :relaxed:

Reply:
On a second thought, I'll defer this. Apart from my annoyance, I don't have a strong argument to make :slightly_smiling_face: